legend-pure-runtime-java-extension-functions/src/test/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/support/function/base/string/TestSubstringCompiled.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.pure.runtime.java.compiled.generation.processors.support.function.base.string;
+
+import org.finos.legend.pure.m3.execution.FunctionExecution;
+import org.finos.legend.pure.m3.tests.AbstractPureTestWithCoreCompiled;
+import org.finos.legend.pure.m3.tests.function.base.string.AbstractTestSubstring;
+import org.finos.legend.pure.m3.tests.function.base.string.AbstractTestToString;
+import org.finos.legend.pure.runtime.java.compiled.execution.FunctionExecutionCompiledBuilder;
+import org.finos.legend.pure.runtime.java.compiled.factory.JavaModelFactoryRegistryLoader;
+import org.junit.After;
+import org.junit.BeforeClass;
+
+public class TestSubstringCompiled extends AbstractTestSubstring
+{
+    @BeforeClass
+    public static void setUp()
+    {
+        AbstractPureTestWithCoreCompiled.setUpRuntime(getFunctionExecution(), JavaModelFactoryRegistryLoader.loader());
+    }
+
+    @After
+    public void cleanRuntime()
+    {
+        AbstractPureTestWithCoreCompiled.runtime.delete("substring.pure");
+    }
```

Comment:
It's important to compile after deleting. Otherwise, you're not fully cleaning up the runtime. Also, it seems to me like this method could be promoted to AbstractTestSubstring.
```suggestion
    @After
    public void cleanRuntime()
    {
        runtime.delete("substring.pure");
        runtime.compile();
    }
```